google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/apis/ServiceAccountKeyUltimateDisplayDialog.java

```diff
@@ -0,0 +1,269 @@
+import com.intellij.execution.runners.ProgramRunner;
+import com.intellij.execution.util.EnvironmentVariable;
+import com.intellij.javaee.run.configuration.RunnerSpecificLocalConfigurationBit;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.ui.BooleanTableCellEditor;
+import com.intellij.ui.BooleanTableCellRenderer;
+import com.intellij.ui.TableUtil;
+import com.intellij.ui.table.JBTable;
+import com.intellij.util.ui.UIUtil;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.stream.IntStream;
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.ListSelectionModel;
+import javax.swing.border.Border;
+import javax.swing.border.EmptyBorder;
+import javax.swing.table.DefaultTableCellRenderer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Dialog visible only in Ultimate Edition that (1) confirms the download of the service account
+ * JSON key for the Google Cloud Libraries , (2) provides information on how to set the environment
+ * variables for local run and (3) allows the user to select App Engine Standard run configurations
+ * to automatically update with these environment variables.
+ */
+public class ServiceAccountKeyUltimateDisplayDialog extends DialogWrapper {
+
+  private final Project project;
+  private final CloudProject cloudProject;
+  private final String downloadPath;
+  private JPanel mainPanel;
+  private ServiceAccountKeyDownloadedPanel commonPanel;
+  private JTable runConfigurationTable;
+  private JLabel runConfigurationUpdateLabel;
+  private JScrollPane scrollPane;
+  private BooleanTableModel<RunnerAndConfigurationSettings> runConfigurationTableModel;
+
+  @VisibleForTesting public static List<RunnerAndConfigurationSettings> configurationSettingsList;
+
+  public ServiceAccountKeyUltimateDisplayDialog(
+      @Nullable Project project, @NotNull CloudProject cloudProject, @NotNull String downloadPath) {
+    super(project);
+    this.project = project;
+    this.cloudProject = cloudProject;
+    this.downloadPath = downloadPath;
+    init();
+    setTitle(GctBundle.message("cloud.apis.service.account.key.downloaded.title"));
+    runConfigurationTable.setTableHeader(null);
+
+    if (runConfigurationTableModel.getRowCount() == 0) {
+      runConfigurationUpdateLabel.setVisible(false);
+      scrollPane.setVisible(false);
+      runConfigurationTable.setVisible(false);
+    }
+  }
+
+  @Nullable
+  @Override
+  protected JComponent createCenterPanel() {
+    return mainPanel;
+  }
+
+  @NotNull
+  @Override
+  protected Action[] createActions() {
```

Comment:
I don't think we need to override this. Can't we just use the standard actions of DialogWrapper? In the same way the `CloudApiManagementConfirmationDialog` is doing it, for example.

Reply:
I see your point. But at least for the Ultimate version, a 'cancel' button might be nice (which would mean intuitively that they don't want to add any env vars instead of having to uncheck them one by one). Also, I think we can get rid of the "apply" altogether.